### PR TITLE
Handle EKF - GPS corner case

### PR DESCRIPTION
**Pull request by mswingtra**
As discussed, this is temporal solution to  #3087. The ALTCTL will be rejected if there is no good local position estimate, which is equivalent to good gps. 

Additionally, I have changed fallback conditions if auto, position or altitude modes are rejected. Earlier it was going back straight to manual which is not optimal and could cause potentially dangerous situation. Now, it should fallback to the previous low-level mode.

### CODE REVIEW

**Comment by LorenzMeier** [comment]
Could you please rebase this instead of merging? @boosfelm It might make sense for Wingtra to review the internal git workflow to avoid merge commits (which in turn are nasty if you need to go back to the history).

If you don't know how to rebase don't worry, in that case I would just cherry-pick those four commits.

File: src/modules/commander/state_machine_helper.cpp [hunk 1]
```diff
@@ -310,11 +310,9 @@ main_state_transition(struct vehicle_status_s *status, main_state_t new_main_sta
 		break;
 
 	case vehicle_status_s::MAIN_STATE_ALTCTL:
-		/* need at minimum altitude estimate */
-		/* TODO: add this for fixedwing as well */
-		if (!status->is_rotary_wing ||
```
Comment (LorenzMeier) [inline src/modules/commander/state_machine_helper.cpp]: A lot of people are actually flying multicopter without GPS, because holding altitude is hard. The longer I look at this the more I have the impression the right fix is to prevent arming.

**Comment by mswingtra** [comment]
Frankly, I'm not sure how to rebase.

File: src/modules/commander/state_machine_helper.cpp (hunk 1 above)
Comment (mswingtra) [inline src/modules/commander/state_machine_helper.cpp]: The solution with circuit breaker is inconvenient for people who fly with GPS and perform tests indoor as well (without vicon). It adds up a parameter that has to be checked and forgetting about it could potentially lead to the same corner case.

**Comment by jonbinney** [comment]
@mswingtra you can achieve the same affect of the rebase by using cherry-pick:
```
git fetch master
git checkout -b gps_failure_rebased origin/master
git cherry-pick 9d5ea4b
(follow instructions to fix any conflicts)
git cherry-pick 0112965
(follow instructions to fix any conflicts)
git cherry-pick 24f5388
(follow instructions to fix any conflicts)
```
After that you'll have a new branch called `gps_failure_rebased` with only your commits on top of the current master. At that point you can either close this PR and open a new one from that branch, or force-push the new branch and overwrite the branch for this PR in place:
```
git push -f wingtra gps_failure_rebased:gps_failure
```

**Comment by LorenzMeier** [comment]
We will address this in EKF2.
